crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -1386,7 +1406,24 @@ pub async fn url_resolve(
     .await
 }
 
-async fn handle_resolve_plugins(
+async fn handle_before_resolve_plugins(
+    lookup_path: Vc<FileSystemPath>,
+    reference_type: Value<ReferenceType>,
+    request: Vc<Request>,
+    options: Vc<ResolveOptions>,
+) -> Result<Option<Vc<ResolveResult>>> {
+    for plugin in &options.await?.before_resolve_plugins {
+        if let Some(result) = *plugin
+            .before_resolve(lookup_path, reference_type.clone(), request)
```

Comment:
That might have a negative performance effect, since it calls the plugin for every resolving in the application. So something like with the resolve plugins would be beneficial: `before_resolve_condition`